xchange-binance/src/main/java/org/knowm/xchange/binance/BinanceExchange.java

```diff
@@ -120,29 +120,35 @@ public void remoteInit() {
                         because their API returns current fee directly */
                   stepSize,
                   null));
-          if (!currencies.containsKey(currentCurrencyPair.base)) {
-            AssetDetail baseAsset = assetDetailMap.get(currentCurrencyPair.base.getCurrencyCode());
-            BigDecimal baseWithdrawlFee =
-                baseAsset != null ? baseAsset.getWithdrawFee().stripTrailingZeros() : null;
-            currencies.put(
-                currentCurrencyPair.base, new CurrencyMetaData(basePrecision, baseWithdrawlFee));
-          }
-          if (!currencies.containsKey(currentCurrencyPair.counter)) {
-            AssetDetail counterAsset =
-                assetDetailMap.get(currentCurrencyPair.counter.getCurrencyCode());
-            BigDecimal counterWithdrawlFee =
-                counterAsset != null ? counterAsset.getWithdrawFee().stripTrailingZeros() : null;
-            currencies.put(
-                currentCurrencyPair.counter,
-                new CurrencyMetaData(counterPrecision, counterWithdrawlFee));
-          }
+
```

Comment:
move the declaration in line 83 `Map<String, AssetDetail> assetDetailMap = accountService.getAssetDetails();` to here

Reply:
Hey @walec51 

`Map<String, AssetDetail> assetDetailMap = accountService.getAssetDetails();` is an API call to get all the assets and should only be called once. Placing it on line 123 put it within the for loop which will call it over and over again resulting in a possible ban due to exceeding the rate limit.

Reply:
ok, your right, didn't notice the loop above